pkg/filesystem/filesystem.go

```diff
@@ -3,14 +3,44 @@ package filesystem
 import (
 	"io/ioutil"
 	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/giantswarm/microerror"
 )
 
-type Store struct{}
+type Store struct {
+	Dir string
+}
 
 func (s *Store) ReadFile(path string) ([]byte, error) {
-	return os.ReadFile(path)
+	if filepath.IsAbs(path) {
+		panic(microerror.Maskf(
+			executionFailedError,
+			"%q is an absolute path; expected sub-path of %q",
+			path, s.Dir,
+		))
+	}
+	return os.ReadFile(filepath.Join(s.Dir, path))
 }
 
 func (s *Store) ReadDir(path string) ([]os.FileInfo, error) {
-	return ioutil.ReadDir(path)
+	if filepath.IsAbs(path) {
+		panic(microerror.Maskf(
+			executionFailedError,
+			"%q is an absolute path; expected sub-path of %q",
+			path, s.Dir,
+		))
+	}
+	return ioutil.ReadDir(filepath.Join(s.Dir, path))
+}
+
+func (s *Store) Version() (string, error) {
```

Comment:
I think adding godoc here may be beneficial. What this method is doing is rather unclear from the signature.